diff --git a/wled00/FX.h b/wled00/FX.h
index bcbab69a..774bb569 100644
--- a/wled00/FX.h
+++ b/wled00/FX.h
@@ -512,11 +512,11 @@ class Segment {
       #ifndef WLED_SAVE_RAM
       CRGBPalette16 _palT;                // temporary palette (slowly being morphed from old to new)
       #endif
-      uint16_t      _dur;                 // duration of transition in ms
+      uint32_t      _dur;                 // duration of transition in ms
       uint16_t      _progress;            // transition progress (0-65535); pre-calculated from _start & _dur in updateTransitionProgress()
       uint8_t       _prevPaletteBlends;   // number of previous palette blends (there are max 255 blends possible)
       uint8_t       _palette, _bri, _cct; // palette ID, brightness and CCT at the start of transition (brightness will be 0 if segment was off)
-      Transition(uint16_t dur=750)
+      Transition(uint32_t dur=750)
       : _oldSegment(nullptr)
       , _start(millis())
       , _colors{0,0,0}
@@ -688,7 +688,7 @@ class Segment {
       */
     inline Segment &markForReset() { reset = true; return *this; }  // setOption(SEG_OPTION_RESET, true)
 
-    void startTransition(uint16_t dur, bool segmentCopy = true);    // transition has to start before actual segment values change
+    void startTransition(uint32_t dur, bool segmentCopy = true);    // transition has to start before actual segment values change
     uint8_t  currentCCT() const; // current segment's CCT (blended while in transition)
     uint8_t  currentBri() const; // current segment's opacity/brightness (blended while in transition)
 
@@ -916,7 +916,7 @@ class WS2812FX {
     inline void fill(uint32_t c) const                        { for (size_t i = 0; i < getLengthTotal(); i++) setPixelColor(i, c); } // fill whole strip with color (inline)
     inline void trigger()                                     { _triggered = true; }  // Forces the next frame to be computed on all active segments.
     inline void setShowCallback(show_callback cb)             { _callback = cb; }
-    inline void setTransition(uint16_t t)                     { _transitionDur = t; } // sets transition time (in ms)
+    inline void setTransition(uint32_t t)                     { _transitionDur = t; } // sets transition time (in ms)
     inline void appendSegment(uint16_t sStart=0, uint16_t sStop=30, uint16_t sStartY = 0, uint16_t sStopY = 1)
                                                               { if (_segments.size() < getMaxSegments()) _segments.emplace_back(sStart,sStop,sStartY,sStopY); }
     inline void suspend()                                     { _suspend = true; }    // will suspend (and canacel) strip.service() execution
@@ -959,7 +959,7 @@ class WS2812FX {
     inline uint16_t getFrameTime() const    { return _frametime; }        // returns amount of time a frame should take (in ms)
     inline uint16_t getMinShowDelay() const { return MIN_FRAME_DELAY; }   // returns minimum amount of time strip.service() can be delayed (constant)
     inline uint16_t getLength() const       { return _length; }           // returns actual amount of LEDs on a strip (2D matrix may have less LEDs than W*H)
-    inline uint16_t getTransition() const   { return _transitionDur; }    // returns currently set transition time (in ms)
+    inline uint32_t getTransition() const   { return _transitionDur; }    // returns currently set transition time (in ms)
     inline uint16_t getMappedPixelIndex(uint16_t index) const {           // convert logical address to physical
       if (index < customMappingSize && (realtimeMode == REALTIME_MODE_INACTIVE || realtimeRespectLedMaps)) index = customMappingTable[index];
       return index;
@@ -1033,7 +1033,7 @@ class WS2812FX {
 
     uint8_t  _brightness;
     uint16_t _length;
-    uint16_t _transitionDur;
+    uint32_t _transitionDur;
 
     uint16_t _frametime;
     uint16_t _cumulativeFps;
diff --git a/wled00/FX_fcn.cpp b/wled00/FX_fcn.cpp
index 3f357daa..1c04954c 100644
--- a/wled00/FX_fcn.cpp
+++ b/wled00/FX_fcn.cpp
@@ -281,7 +281,7 @@ CRGBPalette16 &Segment::loadPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
 }
 
 // starting a transition has to occur before change so we get current values 1st
-void Segment::startTransition(uint16_t dur, bool segmentCopy) {
+void Segment::startTransition(uint32_t dur, bool segmentCopy) {
   if (dur == 0 || !isActive()) {
     if (isInTransition()) _t->_dur = 0;
     return;
@@ -320,7 +320,7 @@ void Segment::startTransition(uint16_t dur, bool segmentCopy) {
     } else {
       DEBUGFX_PRINTF_P(PSTR("-- Started transition without old segment: S=%p T(%p)\n"), this, _t);
     }
-  };
+  }
 }
 
 void Segment::stopTransition() {
@@ -331,8 +331,10 @@ void Segment::stopTransition() {
 void Segment::updateTransitionProgress() const {
   if (isInTransition()) {
     _t->_progress = 0xFFFF;
-    unsigned diff = millis() - _t->_start;
-    if (_t->_dur > 0 && diff < _t->_dur) _t->_progress = diff * 0xFFFFU / _t->_dur;
+    unsigned long diff = millis() - _t->_start;
+    if (_t->_dur > 0 && diff < _t->_dur) {
+      _t->_progress = diff * 0xFFFFU / _t->_dur;
+    }
   }
 }
 
diff --git a/wled00/cfg.cpp b/wled00/cfg.cpp
index e30be759..11381fd0 100644
--- a/wled00/cfg.cpp
+++ b/wled00/cfg.cpp
@@ -520,8 +520,8 @@ bool deserializeConfig(JsonObject doc, bool fromFS) {
   NeoGammaWLEDMethod::calcGammaTable(gammaCorrectVal); // fill look-up tables
 
   JsonObject light_tr = light["tr"];
-  int tdd = light_tr["dur"] | -1;
-  if (tdd >= 0) transitionDelay = transitionDelayDefault = tdd * 100;
+  int64_t tdd = light_tr["dur"] | -1;
+  if (tdd >= 0) transitionDelay = transitionDelayDefault = (uint32_t)tdd * 100U;
   strip.setTransition(transitionDelayDefault);
   CJSON(randomPaletteChangeTime, light_tr[F("rpc")]);
   CJSON(useHarmonicRandomPalette, light_tr[F("hrp")]);
diff --git a/wled00/playlist.cpp b/wled00/playlist.cpp
index 2e51503e..b80fca01 100644
--- a/wled00/playlist.cpp
+++ b/wled00/playlist.cpp
@@ -7,7 +7,7 @@
 typedef struct PlaylistEntry {
   uint8_t preset; //ID of the preset to apply
   uint16_t dur;   //Duration of the entry (in tenths of seconds)
-  uint16_t tr;    //Duration of the transition TO this entry (in tenths of seconds)
+  uint32_t tr;    //Duration of the transition TO this entry (in tenths of seconds)
 } ple;
 
 static byte           playlistRepeat = 1;        //how many times to repeat the playlist (0 = infinitely)
diff --git a/wled00/wled.h b/wled00/wled.h
index 66b33740..acf19bbd 100644
--- a/wled00/wled.h
+++ b/wled00/wled.h
@@ -605,8 +605,8 @@ WLED_GLOBAL uint8_t paletteBlend _INIT(0);        // determines blending and wra
 // transitions
 WLED_GLOBAL uint8_t       blendingStyle            _INIT(0);      // effect blending/transitionig style
 WLED_GLOBAL bool          transitionActive         _INIT(false);
-WLED_GLOBAL uint16_t      transitionDelay          _INIT(750);    // global transition duration
-WLED_GLOBAL uint16_t      transitionDelayDefault   _INIT(750);    // default transition time (stored in cfg.json)
+WLED_GLOBAL uint32_t      transitionDelay          _INIT(750);    // global transition duration
+WLED_GLOBAL uint32_t      transitionDelayDefault   _INIT(750);    // default transition time (stored in cfg.json)
 WLED_GLOBAL unsigned long transitionStartTime;
 WLED_GLOBAL bool          jsonTransitionOnce       _INIT(false);  // flag to override transitionDelay (playlist, JSON API: "live" & "seg":{"i"} & "tt")
 WLED_GLOBAL uint8_t       randomPaletteChangeTime  _INIT(5);      // amount of time [s] between random palette changes (min: 1s, max: 255s)
